test/integration/model/json.test.js

```diff
@@ -227,7 +227,7 @@ describe(Support.getTestDialectTeaser('Model'), function () {
           return this.Event.findAll({
             where: {
               data: {
-                employment: null
+                employment: current.dialect.name === 'mysql' ? 'null' : null
```

Comment:
If we are inserting a SQL NULL we will get actual NULL back, our JSON processing is somehow turning `null` to `'null'`. If we fix that this problem will not appear